appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -2,448 +2,421 @@ package nu.yona.server
-			response.responseData._embedded.buddyConnectRequestMessages[0]._links.self.href.startsWith(bobDunnURL + appService.DIRECT_MESSAGES_PATH_FRAGMENT)
-			bobDunnBuddyMessageAcceptURL.startsWith(response.responseData._embedded.buddyConnectRequestMessages[0]._links.self.href)
+		cleanup:
+			appService.deleteUser(richard)
+			// TODO: How to delete the invited user?
 	}
 
-	def 'Bob accepts Richard\'s buddy request'(){
+	def 'Hacking attempt: Try to get a normal user with a temp password'()
+	{
 		given:
+			def richard = addRichard()
 
 		when:
-			def response = appService.postMessageActionWithPassword(bobDunnBuddyMessageAcceptURL, """{
-				"properties":{
-					"message":"Yes, great idea!"
-				}
-			}""", bobDunnPassword)
+			def response = appService.getResource(richard.url, [:], ["tempPassword": "hack", "includePrivateData": "true"])
 
 		then:
-			response.status == 200
-			response.responseData.properties.status == "done"
-	}
-
-	def 'Richard checks his anonymous messages'(){
-		given:
-
-		when:
-			def response = appService.getAnonymousMessages(richardQuinURL, richardQuinPassword)
-			if (response.responseData._embedded && response.responseData._embedded.buddyConnectResponseMessages) {
-				richardQuinBuddyMessageProcessURL = response.responseData._embedded.buddyConnectResponseMessages[0]._links.process.href
-			}
+			response.status == 400
+			// response.responseData.code == "TODO" // Why do we not get a response code here?
 
-		then:
-			response.status == 200
-			response.responseData._embedded.buddyConnectResponseMessages[0].user.firstName == "Bob ${timestamp}"
-			response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href.startsWith(richardQuinURL + appService.ANONYMOUS_MESSAGES_PATH_FRAGMENT)
-			richardQuinBuddyMessageProcessURL.startsWith(response.responseData._embedded.buddyConnectResponseMessages[0]._links.self.href)
+		cleanup:
+			appService.deleteUser(richard)
 	}
 
-	def 'Richard processes Bob\'s buddy acceptance'(){
+	def 'Hacking attempt: Try to update a normal user with a temp password'()
+	{
 		given:
+			def richard = addRichard()
 
 		when:
-			def response = appService.postMessageActionWithPassword(richardQuinBuddyMessageProcessURL, """{
-				"properties":{
-				}
-			}""", richardQuinPassword)
+			def response = appService.updateResource(richard.url, """{
+				"firstName":"Richard",
+				"lastName":"Quin",
+				"nickname":"RQ",
+				"mobileNumber":"+$timestamp",
+				"devices":[
+					"Nexus 6"
+				],
+				"goals":[
+					"news"
+				]
+			}""", ["Yona-Password": "New password"], ["tempPassword": "hack"])
 
 		then:
-			response.status == 200
-			response.responseData.properties.status == "done"
-	}
-
-	def 'Bob\'s user data will contain Richard as buddy'(){
-		given:
-
-		when:
-			def response = appService.getUser(bobDunnURL, true, bobDunnPassword)
+			response.status == 400
+			// response.responseData.code == "TODO" // Why do we not get a response code here?
```

Comment:
That's not the case (searched for it). I've logged YD-137 for this and mentioned it in the TODOs